### PR TITLE
feat: Attempt to initialize client if uninitialized

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s)
Fixes KOL-5272

### What change does this PR introduce and why?
Updates `kolena_initialized` wrapper to attempt to initialize the client if uninitialized instead of immediately raising an error.

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Comment by consoull** [comment]
@gordonhart we have to hold off on updating examples to drop `kolena.initialize()` until we have a release with this change so we can pin the minimum version

File: kolena/_utils/state.py [hunk 1]
```diff
@@ -129,21 +119,17 @@ def get_client_state() -> _ClientState:
...
 
 
-def is_client_initialized() -> bool:
-    try:
-        get_client_state().assert_initialized()
-    except (InvalidClientStateError, UninitializedError):
-        return False
-    return True
+def is_client_uninitialized() -> bool:
+    return get_client_state().jwt_token is None
```
Comment (gordonhart) [inline kolena/_utils/state.py:123]: Should this not follow the same logic in the removed `assert_initialized`?

Comment (consoull) [reply kolena/_utils/state.py:123]: I view the `base_url` guard as dead code, we provide a default in the constructor (probably shouldn't be an optional) https://github.com/kolenaIO/kolena/blob/trunk/kolena/_utils/state.py#L54.
What really signals the client is initialized is the presence of a jwt token, the auth token is a required to retrieve the jwt token https://github.com/kolenaIO/kolena/blob/trunk/kolena/_utils/state.py#L186

File: kolena/_utils/state.py [hunk 2]
```diff
@@ -129,21 +119,17 @@ def get_client_state() -> _ClientState:
...
 def kolena_initialized(func: Callable) -> Callable:
-    """Raises InvalidKolenaStateError if not initialized"""
+    """Attempts to initialize client if not initialized"""
 
     @functools.wraps(func)
     def wrapper(*args: Any, **kwargs: Any) -> Any:
-        client_state = get_client_state()
-        client_state.assert_initialized()
+        if is_client_uninitialized():
+            kolena.initialize(verbose=True)
```
Comment (ly-kolena) [inline kolena/_utils/state.py]: Should we use `verbose=True` here?

File: kolena/_utils/state.py [hunk 3]
```diff
@@ -129,21 +119,17 @@ def get_client_state() -> _ClientState:
     return CLIENT_STATE.get(_client_state)
 
 
-def is_client_initialized() -> bool:
```
Comment (ly-kolena) [inline kolena/_utils/state.py:132]: Nit: from a class' public interface, `is_client_initialized` reads more natural than `is_client_uninitialized`.

Comment (consoull) [reply kolena/_utils/state.py:132]: We only use the function to check if the client is uninitialized eg. `if not is_client_initialized()`. I prefer `if is_client_uninitialized()` but don't feel strongly

File: kolena/_utils/state.py (hunk 2 above)
Comment (consoull) [reply kolena/_utils/state.py]: Ya I think we should use default args. Will update

**Review by ly-kolena — APPROVED** [review]

**Pull request merged**